lldb/tools/lldb-dap/DAP.cpp

```diff
@@ -85,6 +85,105 @@ const char DEV_NULL[] = "/dev/null";
 
 namespace lldb_dap {
 
+static DAPSessionManager *instance = nullptr;
```

Comment:
This can go inside DAPSessionManager::GetInstance(). Also add a commnt before it:
```
// NOTE: intentional leak to avoid issues with C++ destructor chain
```